test/system/test_cvp_change_control_api.py

```diff
@@ -0,0 +1,784 @@
+
+    def test_api_change_control_approval_get_one(self):
+        """ Verify change_control_approval_get_one
+         """
+        pprint(
+            "test_api_change_control_approval_get_one")
+        if self.get_version():
+            # Create task
+            task_id = self.create_task()
+
+            # Create change control
+            self.create_change_control_for_tasks(
+                task_id)
+
+            # Approve change control
+            self.approve_change_control()
+
+            pprint('APPROVAL GET ONE...')
+            approval_get_one = self.api.change_control_approval_get_one(
+                self.cc_id)
+            assert approval_get_one is not None
+            assert approval_get_one['value']['approve']['value'] is True
+            assert approval_get_one['value']['approve']['notes'] == APPROVE_NOTE
+            assert approval_get_one['value']['key']['id'] == self.cc_id
+            time.sleep(1)
+
+            # verify with actual api output
+
+            response = self.get_cc_status(
+                self.cc_id)
+
+            assert response is not None
+            assert approval_get_one['value']['key']['id'] == response['value']['key']['id']
+            assert approval_get_one['value']['approve'][
+                'notes'] == response['value']['approve']['notes']
+            assert approval_get_one['value']['approve'][
+                'value'] == response['value']['approve']['value']
+
+            # Delete change control
+            self.delete_change_control(self.cc_id)
+
+            # Cancel Task
+            self.cancel_task(task_id)
+
+    def test_api_change_control_approval_get_one_without_approve(self):
+        """ Verify change_control_approval_get_one_without_approve
+         """
+        pprint(
+            "test_api_change_control_approval_get_one_without_approve")
+        if self.get_version():
+            # Create task
+            task_id = self.create_task()
+
+            # Create CC
+            self.create_change_control_for_tasks(
+                task_id)
+
+            pprint(
+                'APPROVAL GET ONE WITHOUT APPROVE...')
+            approval_get_one = self.api.change_control_approval_get_one(
+                self.cc_id)
+            assert approval_get_one is None
+
+            # Delete CC
+            self.delete_change_control(self.cc_id)
+
+            # Cancel Task
+            self.cancel_task(task_id)
+
+    def test_api_change_control_create_for_empty_tasks_list(self):
+        """ Verify change_control_create_for_tasks for empty task list
+        """
+        pprint(
+            "test_api_change_control_create_for_empty_tasks_list")
+        if self.get_version():
+            # Set client apiversion if it is not already set
+            pprint(
+                'RUN TEST FOR V3 CHANGE CONTROL APIs')
+            with self.assertRaises(CvpRequestError):
+                self.create_change_control_for_tasks(
```

Comment:
Why send an empty list fo create_change_control_for_tasks if the parameter is intended to be a single task_id? Within create_change_control_for_tasks the task_id parameter is wrapped in a list before being sent to the api function.

Reply:
This is a negative test case we have added by passing invalid input as empty task_id.

Reply:
We have tried to cover the negative test cases for most of the invalid values, to make sure APIs are properly responding in case of invalid values. But we can remove it if it does not make sense.